pieforproviders/pieforproviders
# [93] User confirmation after signup

Adds devise views and custom mailer (left in the ones we're not using now because I'm sure we'll need them later), overrides devise confirmation controller to return JSON (and specifically the token), helper rake task for testing/linting/rswag generation before PR, non-rendering React component to make the API call to the confirmation backend, error handling and translation on the front-end

# 💅🏼 What issue does this fix?
#93 

## 🍂 Before You Submit


## 🛷 Deployment Considerations


## 🧵 Steps to set up locally

restart your rails server; might need to restart the frontend too but I don't think so.

## 🧳 Steps to test
- sign up as a new user
- click the confirmation link that opens in the browser
- you should be logged in and redirected to getting-started
- log out
- click the confirmation link again
- you should get an error telling you your user has already been confirmed, and you should try logging in
- log in as the new user
- you should be logged in and redirected to getting-started
- visit `localhost:3000/confirm?confirmation_token=cactus`
- you should get an error indicating your token is invalid and that you should request a new confirmation token (this behavior does not exist)

## 🕯 Caveats, concerns
We've got error handling for expired links but no tests for it, I'd like to write a ticket for it instead of holding this PR up any longer.

Ditto for requesting a new confirmation token.
